### ISSUE TITLE
MqttRpcClient.ExecuteRpcAsync no longer works if MqttClient already has a message receive handler

### ISSUE THREAD

**Issue by BrianG500** [chkr1011/MQTTnet#1093]
### Describe the bug
After updating from 3.0.12 to 3.0.14, MqttRpcClient.ExecuteRpcAsync no longer works properly if underlying MqttClient has its own MqttApplicationMessageReceivedHandler.

RpcAwareApplicationMessageReceivedHandler looks like it was changed to return immediately after OriginalHandler instead of also calling MqttRpcClient's handler.

### Which project is your bug related to?
- MQTTnet.Extensions.Rpc

### To Reproduce
1. Using version MQTTnet '3.0.14'
2. Create a MqttClient or ManagedMqttClient
3. Add ApplicationMessageReceivedHandler handler to the client
4. Create a MqttRpcClient using this same client
5. Run 'ExecuteRpcAsync'
6. MqttClient message received handler is called
7. MqttRpcClient message handler is skipped
8. ExecuteRpcAsync fails

### Expected behavior
- MqttRpcClient.ExecuteRpcAsync should work if using a MqttClient or ManagedMqttClient that already has its own message receive handler.
- Verified that RPC_Tests only work because the MqttRpcClient is using a response MqttClient without its own ApplicationMessageReceivedHandler so they do not exercise the OriginalHandler condition.

### Screenshots
Bug appears to have been introduced with this change from awaits to returns...
![image](https://user-images.githubusercontent.com/2412587/107951145-7c1f4d80-6f5d-11eb-9687-6c3cf1ddcc94.png)

### Proposed Fix: ExecuteRpcAsynce works properly again by simply removing the immediate return after calling OriginalHandler
.
```
public Task HandleApplicationMessageReceivedAsync(MqttApplicationMessageReceivedEventArgs eventArgs)
{
    if (OriginalHandler != null)
    {
        OriginalHandler.HandleApplicationMessageReceivedAsync(eventArgs);
    }

    return _handleReceivedApplicationMessageAsync(eventArgs);
}
```

**Comment by SeppPenner** [comment]
You're right, I guess. @chkr1011 please have a look at my pull request when you have time.

**Comment by chkr1011** [comment]
In my opinion the issue is that both handlers must be called in order to work correctly. Currently only one is called and the method returns. I fixed this in master. Now both handlers are invoked properly. Please let me know if this fixes your issue.

**Comment by BrianG500** [comment]
@chkr1011 I agree that both handlers should be called in order. However, I verified that @borigas is right, and believe the RpcAwareApplicationMessageReceivedHandler should be reverted back to an async Task with awaits. It is a subtle thing depending on what your message handlers are doing, but you can verify by...

Modify the RPC_Tests ExecuteSuccess() method with this...

```
var requestSender = await testEnvironment.ConnectClientAsync();
requestSender.ApplicationMessageReceivedHandler = new MqttApplicationMessageReceivedHandlerDelegate(async e =>
{
    for (int i = 0; i < 10; i++)
    {
        Console.WriteLine($"OriginalHandler loop {i}");
        await Task.Delay(50).ConfigureAwait(false);
    }
});
```

and also modify the RpcAwareApplicationMessageReceivedHandler back to async Task...

```
public async Task HandleApplicationMessageReceivedAsync(MqttApplicationMessageReceivedEventArgs eventArgs)
{
    if (OriginalHandler != null)
    {
        Console.WriteLine("Calling OriginalHandler...");
        await OriginalHandler.HandleApplicationMessageReceivedAsync(eventArgs).ConfigureAwait(false);
        Console.WriteLine("OriginalHandler finished.");
    }

    Console.WriteLine("Calling Rpc handler...");
    await _handleReceivedApplicationMessageAsync(eventArgs).ConfigureAwait(false);
}
```

You will then get the desired behavior of executing both handlers to completion in order...

![image](https://user-images.githubusercontent.com/2412587/108405931-5311ed80-71e7-11eb-8f0c-f64cb2d6ca66.png)

If you run this same test with PR change OriginalHandler will not run to completion before test finishes and you will get something like this...

![image](https://user-images.githubusercontent.com/2412587/108406305-da5f6100-71e7-11eb-933d-b0d50577c5a7.png)

**Comment by SeppPenner** [comment]
Let me know if this is fixed as I can close my pull request as invalid then :)

**Comment by chkr1011** [comment]
In my opinion this is fixed. You can close your pull request.

**Comment by BrianG500** [comment]
I haven't had time today to pull down master and test, but I think it is now fixed. Thanks to both of you for the quick response.

**Comment by SeppPenner** [comment]
Ok, thanks :)

**Comment by chkr1011** [comment]
I will close this because it is fixed in 3.0.15-rc2.
